bundles/org.openhab.binding.meater/src/main/java/org/openhab/binding/meater/internal/handler/MeaterBridgeHandler.java

```diff
@@ -0,0 +1,152 @@
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.openhab.binding.meater.internal.MeaterBridgeConfiguration;
+import org.openhab.binding.meater.internal.MeaterException;
+import org.openhab.binding.meater.internal.api.MeaterRestAPI;
+import org.openhab.binding.meater.internal.discovery.MeaterDiscoveryService;
+import org.openhab.binding.meater.internal.dto.MeaterProbeDTO;
+import org.openhab.core.i18n.LocaleProvider;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link MeaterBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Jan Gustafsson - Initial contribution
+ */
+@NonNullByDefault
+public class MeaterBridgeHandler extends BaseBridgeHandler {
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(THING_TYPE_BRIDGE);
+
+    private int refreshTimeInSeconds = 300;
+
+    private final Gson gson;
+    private final HttpClient httpClient;
+    private LocaleProvider localeProvider;
+    private final Map<String, MeaterProbeDTO.Device> meaterProbeThings = new ConcurrentHashMap<>();
+
+    private @Nullable MeaterRestAPI api;
+    private @Nullable ScheduledFuture<?> refreshJob;
+
+    public MeaterBridgeHandler(Bridge bridge, HttpClient httpClient, Gson gson, LocaleProvider localeProvider) {
+        super(bridge);
+        this.httpClient = httpClient;
+        this.gson = gson;
+        this.localeProvider = localeProvider;
+    }
+
+    @Override
+    public void initialize() {
+        MeaterBridgeConfiguration config = getConfigAs(MeaterBridgeConfiguration.class);
+
+        MeaterRestAPI meaterRestAPI = new MeaterRestAPI(config, gson, httpClient, localeProvider);
+        refreshTimeInSeconds = config.refresh;
+
+        if (config.email == null || config.password == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Configuration of username and password is mandatory");
+        } else if (refreshTimeInSeconds < 0) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Refresh time cannot be negative!");
+        } else {
+            try {
+                this.api = meaterRestAPI;
+                api.login();
+                scheduler.execute(() -> {
+                    updateStatus(ThingStatus.UNKNOWN);
+                    startAutomaticRefresh();
+                });
+            } catch (MeaterException | RuntimeException e) {
```

Comment:
Why is it needed to catch `RuntimeException` here?

Reply:
I'll remove the try&catch.